src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1311 @@
+            x1
+            sage: S = LPASeed({x1:2},base_ring=ZZ)
+            sage: T = S.mutate(0, inplace=False)
+            sage: T==S
+            False
+        """
+
+        # we check if cluster variables are is_equivalent of each other,
+        # up to unit multiple. Since clusters determine the seed,
+        # this is sufficient
+        return self.is_equivalent(other)
+
+    def _copy_(self):
+        r"""
+        Returns a copy of ``self``.
+
+        TESTS::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: T1 = S._copy_()
+            sage: T1.mutate(0, inplace=True)
+            A seed with cluster variables [(x2 + 1)/x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+            sage: T2 = S.mutate(0, inplace=False)
+            sage: T1==T2
+            True
+        """
+        return LPASeed(self)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        TESTS::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: print(LPASeed({x1: 1 + x2, x2: 1 + x1}))
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+        """
+        return ("A seed with cluster variables {0}"
+                " and exchange polynomials {1}").format(self.cluster(),
+                                                        self.exchange_polys())
+
+    def __hash__(self):
+        """
+        Return a hash of ``self``.
+
+        EXAMPLES:
+
+        We check that two seeds with the same hash are equal::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: T = LPASeed(S)
+            sage: S.mutate([0,1,0])
+            A seed with cluster variables [(x1 + 1)/x2, (x1 + x2 + 1)/(x1*x2)] and exchange polynomials [x2 + 1, x1 + 1]
+            sage: T.mutate([1,0])
+            A seed with cluster variables [(x1 + x2 + 1)/(x1*x2), (x1 + 1)/x2] and exchange polynomials [x2 + 1, x1 + 1]
+            sage: hash(S) == hash(T)
+            True
+            sage: S == T
+            True
+        """
+        return hash(frozenset(self.cluster()))
+
+    def _latex_(self):
+        r"""
+        Returns a `\LaTeX` representation of this seed.
+
+        TESTS::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: latex(LPASeed({x1: 1 + x2, x2: 1 + x1}))
+            \left(\left(x_{1}, x_{2} + 1\right), \left(x_{2}, x_{1} + 1\right)\right)
+
+        """
+        cvars = self.cluster()
+        polys = self.exchange_polys()
+        cvar_poly_pairs = list()
+        for i in range(self.rank()):
+            cvar_poly_pairs.append(latex(tuple([cvars[i], polys[i]])))
+
+        return latex(tuple(cvar_poly_pairs))
+
+    def are_laurent_polys_trivial(self):
+        r"""
+        Returns whether this seed has Laurent polynomials with nontrivial denominators.
```

Comment:
```suggestion
        Return whether this seed has Laurent polynomials with nontrivial denominators.
```
This method also needs doctests.